### DISCUSSION TITLE
218- V/asymptote fails during the installation. How to resolve this issue?

### DISCUSSION THREAD

**Discussion by Anjalijhaa** [neperfepx/neper#785] [Q&A] [answered]
Even after installing the asymptote, the ctest for the asymptote fails.

The following tests failed:
218- V/asymptote (Failed)
Errors while running CTest

**Comment by rquey** [comment]
Did you also install Tex Live? https://neper.info/doc/introduction.html#fine-configuration

**Reply by Anjalijhaa** [reply]
Yes, I did

**Reply by rquey** [reply]
Run the following and report any error:
```
neper -T -n 2
neper -V n2-id1.tess -space pf -print test
```

**Reply by Anjalijhaa** [reply]
I ran the edited commands. It is showing the following error
```
======================== N e p e r =======================

Info : A software package for polycrystal generation and meshing.

Info : Version 4.7.1-5

Info : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)

Info : Running on 9 threads.

Info : <https://neper.info/>

Info : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.

Info : No initialization file found (`/home/anjali/.neperrc').

Info : ---------------------------------------------------------------

Info : MODULE -V loaded with arguments:

Info : [ini file] (none)

Info : [com line] n2-id1.tess -space pf -print test

Info : ---------------------------------------------------------------

Info : - Reading arguments...

Info : Loading tessellation...

Info : [i] Parsing file `n2-id1.tess'...

Info : [i] Parsed file `n2-id1.tess'.

Info : Printing pole figure...

Info : - Computing symbols...

Info : - Generating png file (1200x900 pixels)...

Info : [o] Writing file `test.png'...

convert-im6.q16: attempt to perform an operation not allowed by the security policy `PDF' @ error/constitute.c/IsCoderAuthorized/421.

convert-im6.q16: no images defined `png:test.png' @ error/convert.c/ConvertImageCommand/3229.

_shipout(prefix,f,currentpatterns,format,wait,view,t);

^

/usr/share/asymptote/plain_shipout.asy: 116.11: runtime: shipout failed

Info : [o] Wrote file `test.png'.

Info : Printing scale...

Info : Elapsed time: 1.033 secs.
```

**Reply by rquey** [reply, marked as answer]
See the orange-highlighted note regarding `/etc/ImageMagick-6/policy.xml` in https://neper.info/doc/introduction.html#fine-configuration.

**Reply by Anjalijhaa** [reply]
The comments in the orange-highlighted note are already present in the /etc/ImageMagick-6/policy.xml file.

**Reply by rquey** [reply]
But did you make the changes? If so, you'll have to Google for a solution...

**Comment by sajutabraham** [comment]
PLEASE OPEN THE FILE
sudo nano /etc/ImageMagick-6/policy.xml
THEN EDIT THE LINE 
< policy domain="coder" rights="none" pattern="PDF" / >
TO 
< policy domain="coder" rights="read|write" pattern="PDF" / >
SAVE AND EXIT. 
GO TO NEPER INSTALLATION FOLDER
...neper<your version>/src/build
RUN THE COMMAND
ctest -R V/asymptote
SEE THE RESULT, OR YOU MAY ALSO RUN THE ENTIRE TEST USING
ctest

**Reply by Fengyixin-research** [reply]
@sajutabraham It worked. Thank you!
